adapter/ryu/omniui/omniui.py

```diff
@@ -71,19 +85,465 @@ def stats_reply_handler(self, ev):
+
+        mod = parser.OFPFlowMod(datapath=datapath, priority=priority,
+                                match=match, instructions=inst)
+        datapath.send_msg(mod)
+
+    #
+    # try post json to core
+    #
+    def post_json_to_core(self, url, data):
+        try:
+            resp = requests.post(url, data = data, headers = {'Content-Type': 'application/json'})
+            print resp
+        except Exception, e:
+            print(str(e))
+
+    #
+    # handle add switch event
+    #
+    @set_ev_cls(event.EventSwitchEnter)
+    def add_device_handler(self, ev):
+        switch = ev.switch
+        print '*****add device*****'
+
+        # format dpid
+        temp = "%016x" %switch.dp.id
+        temp = map(str, temp)
+        for i in range(2, 23, 3):
+            temp.insert(i, ':')
+
+        addDevice = {
+            'dpid': ''.join(temp),
+            'controller': controllerName
+        }
+
+        self.port_to_feature[addDevice['dpid']] = {}
+
+        print json.dumps(addDevice)
+        self.post_json_to_core("http://localhost:5567/publish/adddevice", json.dumps(addDevice))
+
+        # send add port event
+        for port in switch.ports:
+            n_ev = event.EventPortAdd(port)
+            self.send_event_to_observers(n_ev)
+
+    #
+    # handle delete switch event
+    #
+    @set_ev_cls(event.EventSwitchLeave)
+    def del_device_handler(self, ev):
+        switch = ev.switch
+        print '*****del device*****'
+
+        # format dpid
+        temp = "%016x" %switch.dp.id
+        temp = map(str, temp)
+        for i in range(2, 23, 3):
+            temp.insert(i, ':')
+
+        delDevice = {
+            'dpid': ''.join(temp),
+            'controller': controllerName
+        }
+
+        if delDevice['dpid'] in self.port_to_feature:
+            del self.port_to_feature[delDevice['dpid']]
+
+        print json.dumps(delDevice)
+        self.post_json_to_core("http://localhost:5567/publish/deldevice", json.dumps(delDevice))
+
+    #
+    # handle modify port event
+    #
+    @set_ev_cls(event.EventPortModify)
+    def mod_port_handler(self, ev):
+        port = ev.port
+        print '*****mod port*****'
+
+        if port.is_down():
+            print '***down***'
+
+            # format dpid
+            temp = "%016x" %port.dpid
+            temp = map(str, temp)
+            for i in range(2, 23, 3):
+                temp.insert(i, ':')
+
+            modPort = {
+                'dpid': ''.join(temp),
+                'port': str(port.port_no),
+                'controller': controllerName
+            }
+
+            if modPort['dpid'] in self.port_to_feature:
+                if modPort['port'] in self.port_to_feature[modPort['dpid']]:
+                    del self.port_to_feature[modPort['dpid']][modPort['port']]
+
+            print json.dumps(modPort)
+            self.post_json_to_core("http://localhost:5567/publish/delport", json.dumps(modPort))
```

Comment:
Make core IP address a variable rather than hard coded `localhost` everywhere in `omniui.py`.